src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -2574,4 +2281,370 @@ public static void shutdownBlocking(long timeout, TimeUnit unit) throws Interrup
         ExecutorUtils.shutdownNowAndWait(timeout, unit, syncExecutor);
         resetTidying();
     }
+    
+    private static abstract class Builder
+    {
+        protected final Descriptor descriptor;
+        protected final TableMetadataRef metadataRef;
+        protected final TableMetadata metadata;
+
+        public IndexSummary summary;
+        public DecoratedKey first;
+        public DecoratedKey last;
+        public IFilter bf;
+        public FileHandle ifile;
+        public FileHandle dfile;
+
+        public Builder(Descriptor descriptor, TableMetadataRef metadataRef)
+        {
+            this.descriptor = descriptor;
+            this.metadataRef = metadataRef;
+            this.metadata = metadataRef.get();
+        }
+
+        protected SSTableReader internalOpenSSTable(Set<Component> components,
```

Comment:
If we make the `Factory` a field on the `Builder` itself, `internalOpenSSTable()` might not even pull its weight, and we could just inline the call to `Factory#open()`. Not super important, but something to think about.

Reply:
good idea